src/ConfigInterface.php

```diff
@@ -96,6 +96,15 @@ public function getUsingCache(): bool;
      */
     public function registerCustomFixers(iterable $fixers): self;
 
+    /**
+     * Adds custom rule sets.
+     *
+     * `$ruleSets` must follow `'@RuleName' => RuleClass::class` convention.
+     *
+     * @param array<string, string> $ruleSets
+     */
+    public function registerCustomRuleSets(array $ruleSets): self;
```

Comment:
In order to provide this feature but without breaking backward compatibility we could get rid of this method and keep only `RuleSets::registerRuleSet()`. The usage would be slightly less DX-friendly, but would work:

```php
<?php

RuleSets::registerRuleSet('@MyRuleSet', \MyNameSpace\MyRuleSetClass::class);

$config = new PhpCsFixer\Config();

return $config
    ->setRules([
        '@MyRuleSet' => true,
    ])
;
```

We can also add this method to `PhpCsFixer\Config` only (without providing contract through interface) with `@TODO Introduce it in ConfigInterface in 4.0`. This would allow fluent usage for most of the cases (do people really make their own implementation of `ConfigInterface` 😅?